index.test.js

```diff
@@ -0,0 +1,122 @@
+    })
+
+    // test("/persons returns 404 if no one to list", async () => {
+    //     isPersonsEmpty = true
+    //     const response = await request(app).get("/persons")
+    //     expect(response.statusCode).toBe(404)
+    // })
+})
+
+describe("POST to /persons", () => {
+  describe("given username and height", () => {
+    test("Reject if name is empty", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Test if the username or height is valid", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "testUser",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(200);
+    });
+
+    test("Check if user has more than 4 characteres", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "Mar",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Check if user has less than 50 characteres", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "MarioMarioMarioMarioMarioMarioMarioMarioMarioMarioa",
+        height: "180",
+      });
+      expect(response.statusCode).toBe(400);
+    });
+
+    test("Reject if height is less than 50 cm", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "Mario",
+        height: "49",
+      });
+      expect(response.statusCode).toBe(400);
+    });
+    
+    test("Reject if height is over than 250 cm", async () => {
+      const response = await request(app).post("/persons").send({
+        name: "Mario",
+        height: "251",
+      });
+      expect(response.statusCode).toBe(400);
+    // expect(response.body).toEqual(expect.stringContaining("maximum"))
+    });
+  });
+});
+
+describe("GET to /persons/:user",() => {
+    test("Return user data if user exist", async () => {
+        const response = await request(app).get("/persons/userTest")
+        expect(response.statusCode).toBe(200)
+    })
+    test("Return not found if user doesn't exist", async () => {
+        const response = await request(app).get("/persons/nonExistentUser")
+        expect(response.statusCode).toBe(204)
+    })
+})
+
+describe("PUT to /persons", () => {
+  test("Updating a inexistent user is possible", async () => {
+    const response = await request(app).put(
+      "/persons/nonExistentUser?test=true"
+    );
+    expect(response.statusCode).toBe(404);
+  });
+
+  test("Updating a existent user is possible", async () => {
+    const response = await request(app).put("/persons/userTest?test=true");
+    expect(response.statusCode).toBe(202); 
+    // expect(response.body).toEqual(expect.stringContaining("successfully"))
+  });
+
+  test("Updating a existent user without query return bad query", async () => {
```

Comment:
I am curious why you were thinking about there query param :)

Reply:
to have a error 400 if someone tries to PUT nothing as query
without it, it would return 200

Reply:
A query param is not required

Reply:
I thought I couldn't send body in PUT requests

Reply:
fixed btw